WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductCoordinator.swift

```diff
@@ -0,0 +1,67 @@
+import UIKit
+import Yosemite
+
+/// Controls navigation for the flow to add a product given a navigation controller.
+///
+final class AddProductCoordinator: Coordinator {
+    var navigationController: UINavigationController
+
+    private let siteID: Int64
+    private let sourceView: UIBarButtonItem
+
+    init(siteID: Int64, sourceView: UIBarButtonItem, sourceNavigationController: UINavigationController) {
+        self.siteID = siteID
+        self.sourceView = sourceView
+        self.navigationController = sourceNavigationController
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func start() {
+        presentProductTypeBottomSheet()
+    }
+}
+
+// MARK: Navigation
+private extension AddProductCoordinator {
+    func presentProductTypeBottomSheet() {
+        let title = NSLocalizedString("Select a product type",
+                                      comment: "Message title of bottom sheet for selecting a product type to create a product")
+        let viewProperties = BottomSheetListSelectorViewProperties(title: title)
+        let command = ProductTypeBottomSheetListSelectorCommand(selected: nil) { selectedProductType in
+            self.navigationController.dismiss(animated: true) {
+                // Strong reference to `self` is required since `AddProductCoordinator` is not strongly referenced by any class.
+                self.presentProductForm(productType: selectedProductType)
+            }
```

Comment:
Making the `presentProductForm` form outside the `dismiss` completion block seems to fix the delay problem in my simulator!
```suggestion
self.navigationController.dismiss(animated: true)
self.presentProductForm(productType: selectedProductType)
```

![fast-transitions](https://user-images.githubusercontent.com/562080/92952087-ff73ca80-f424-11ea-838d-e7c6307a82dd.gif)

Reply:
Nice suggestion! it isn't necessary to wait for the dismiss completion to navigate to the product form indeed. updated in: https://github.com/woocommerce/woocommerce-ios/pull/2802/commits/5dd038da898e1407f480e277ede5d555eababe9e